middleware/kubernetes/kubernetes.go

```diff
@@ -92,13 +101,27 @@ func (k *Kubernetes) Services(state request.Request, exact bool, opt middleware.
 		return nil, nil, e
 	}
 
+	if state.Name() == DefaultNSName+"."+r.zone && state.Type() == "A" {
```

Comment:
@johnbelamaric : `strings.Index(state.Name(), DefaultNSName) == 0` would falsely catch an A record request for `ns.dns.svc.cluster.local`... which could be a valid domain name for service "ns" in  namespace "dns".